File: 04-quick-sort/ruby/recursive_max.rb
# frozen_string_literal: true

def recursive_max(arr)
  case arr.size
  when 0
    nil
  when 1
    arr[0]
  else
    sub_max = recursive_max(arr[1..-1])
    sub_max > arr[0] ? sub_max : arr[0]
  end
end

File: 04-quick-sort/python/recursive_max.py
def recursive_max(arr):
    if len(arr) == 0:
        return None
    elif len(arr) == 1:
        return arr[0]
    else:
        sub_max = recursive_max(arr[1::])
        if sub_max > arr[0]:
            return sub_max
        else:
            return arr[0]


